src/lib/components/main-button.svelte

```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+  //Styles passed along to custom Button Component
+  export let styles: string = '';
+  export let disabled: boolean = false;
+  //Function Passed along to component that is used as a on:click
+  export let onClick;
+  //Boolean to state if button is active or not
+  export let active: boolean = false;
+</script>
+
+<button on:click={onClick} class={styles} {disabled} class:active
```

Comment:
You can probably use [event forwarding](https://svelte.dev/docs#Element_directives) here instead of passing the prop through. For example:

```js
             👇
<button on:click class={styles} {disabled} class:active>
  <slot />
</button>
```